www/js/app.js

```diff
@@ -2303,6 +2318,112 @@ function articleLoadedSW (iframeArticleContent) {
+            if (a && a.nodeName === 'A') {
+                // console.debug(event.type, event.target, a);
+                const suppressContextMenuHandler = function (e) {
+                    e.preventDefault();
+                    e.stopPropagation();
+                };
+                // Prevent context menu on this anchor element
+                a.addEventListener('contextmenu', suppressContextMenuHandler, true);
+                if (/touchstart|pointerdown/.test(event.type)) {
+                    a.touched = true; // Used to prevent dismissal of popver on mouseout if initiated by touch
+                }
+                if (a.style.userSelect === undefined) {
+                    // This prevents selection of the text in a touched link in Safari for iOS and Edge Legacy / UWP
+                    a.style.webkitUserSelect = 'none';
+                    a.style.msUserSelect = 'none';
+                }
+                // Check if a popover div is currently being hovered
+                const divs = iframeDoc.getElementsByClassName('kiwixtooltip');
+                let divIsHovered = false;
+                Array.prototype.slice.call(divs).forEach(function (div) {
+                    if (div.matches(':hover')) divIsHovered = true;
+                });
+                // Only add a popover to the link if a current popover is not being hovered (prevents popovers showing for links in a popover)
+                if (!divIsHovered) {
+                    // Prevent text selection while popover is open in modern browsers
+                    a.style.userSelect = 'none';
+                    // Resolve the true app theme
+                    const isDarkTheme = uiUtil.isDarkTheme(params.appTheme);
+                    // Get and populate the popover corresponding to the hovered or focused link
+                    uiUtil.populateKiwixPopoverDiv(event, a, appstate.baseUrl, isDarkTheme, selectedArchive);
+                }
+                const outHandler = function (e) {
+                    // console.debug('outHandler', e.type);
+                    setTimeout(function () {
+                        a.popoverisloading = false;
+                        if (/blur/.test(e.type) || !a.touched) {
+                            uiUtil.removeKiwixPopoverDivs(iframeDoc);
+                            a.touched = false;
+                        }
+                        a.style.webkitUserSelect = 'auto';
+                        a.style.msUserSelect = 'auto';
+                        a.style.userSelect = 'auto';
+                        a.removeEventListener(e.type, outHandler);
+                        a.removeEventListener('contextmenu', suppressContextMenuHandler, true);
+                    }, 250);
+                };
+                if (/mouseover/.test(event.type)) {
+                    a.addEventListener('mouseleave', outHandler);
+                }
+                if (/pointerdown/.test(event.type)) {
```

Comment:
Are these meant to be non-mutually exclusive? That is, not using `else if`? Just to be clear if both of them are true we want to do both actions and attach multiple event listeners?

Reply:
Just want to evoke the cleanup routine, so if some or any of them have been defined, their "opposite" will be used to clean up.

Reply:
It's `pointerdown` and `touchstart` that are mutually exclusive, but I don't think adding and `else` makes any difference, as if one of these doesn't exist its opposite (`pointerup' ,  'touchend' simply never gets attached). I think of this code as a broom sweeping away anything it finds.... 🧹